extensions/functions_datetime.yaml

```diff
@@ -6,25 +6,49 @@ scalar_functions:
     description: Extract portion of a date/time value.
     impls:
       - args:
-          - options: [ YEAR, MONTH, DAY, SECOND ]
+          - options: [ YEAR, ISO_YEAR, QUARTER, MONTH, DAY, DAY_OF_WEEK, ISO_DAY_OF_WEEK, DAY_OF_YEAR, WEEK,
+                       ISO_WEEK, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND, NANOSECOND, SUBSECOND, EPOCH ]
             name: The part of the value to extract.
             required: true
           - value: timestamp
         return: i64
       - args:
-          - options: [ YEAR, MONTH, DAY, SECOND ]
+          - options: [ YEAR, ISO_YEAR, QUARTER, MONTH, DAY, DAY_OF_WEEK, ISO_DAY_OF_WEEK, DAY_OF_YEAR, WEEK,
+                       ISO_WEEK, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND, NANOSECOND, SUBSECOND, EPOCH,
+                       TIMEZONE_OFFSET ]
             name: The part of the value to extract.
             required: true
           - value: timestamp_tz
+          - value: string
```

Comment:
I can see why you might want a timezone string here, but at the very least it must be documented that that is what it is, and which formats it supports. However, I would much rather see the reverse of `assume_timezone`, e.g. `with_timezone(string, timestamp_tz) -> timestamp`, in which case this implementation can just always use UTC, and a user could write `extract(..., with_timezone(timezone, timestamp))` if they want a different one. That way, the existing function signature doesn't need to be changed, and the functions represent more primitive operations (only `assume_timezone` and `with_timezone` would need to understand what timezones are, which is great because timezones are terrible).

Reply:
I'm not sure doing `extract(assume_timezone(timestamp, timezone), component)` would work. As far as I understand extracted components should be local. So timezone would need to be passed to `extract` as well.
e.g. `extract(assume_timezone(timestamp, timezone), component, timezone)`

Reply:
> As far as I understand extracted components should be local

Okay, on second thought, that makes sense. I was thinking in UTC, but `timezone_tz` is supposed to be agnostic of that.